show-device-location-using-indoor-positioning/src/main/java/com/esri/arcgismaps/sample/showdevicelocationusingindoorpositioning/MainActivity.kt

```diff
@@ -0,0 +1,365 @@
+                return
+            }
+            val queryParameters = QueryParameters().apply {
+                // set a limit of 1 on the number of returned features per request
+                maxFeatures = 1
+                // 1=1 is a true where clause which will result in all matching records being returned
+                whereClause = "1 = 1"
+                // find and sort out the orderByFields by most recent first
+                orderByFields.add(
+                    OrderBy(
+                        dateCreatedFieldName,
+                        sortOrder = SortOrder.Descending
+                    )
+                )
+            }
+            lifecycleScope.launch {
+                positioningFeatureTable.queryFeatures(queryParameters)
+                    .onSuccess { queryResults ->
+                        // perform search query using the queryParameters
+                        queryResults.forEach { feature ->
+                            // check if serviceFeatureTable contains positioning data
+                            // The ID that identifies a row in the positioning table.
+                            val globalID =
+                                feature.attributes[positioningFeatureTable.globalIdField].toString()
+                            val positioningId = Guid(globalID)
+                            // Setting up IndoorsLocationDataSource with positioning, pathways tables and positioning ID.
+                            // positioningTable - the "ips_positioning" feature table from an IPS-enabled map.
+                            // pathwaysTable - An ArcGISFeatureTable that contains pathways as per the ArcGIS Indoors Information Model.
+                            // Setting this property enables path snapping of locations provided by the IndoorsLocationDataSource.
+                            // levelsTable - An ArcGISFeatureTable that contains floor levels in accordance with the ArcGIS Indoors Information Model.
+                            // Providing this table enables the retrieval of a location's floor level ID.
+                            // positioningID - an ID which identifies a specific row in the positioningTable that should be used for setting up IPS.
+                            mIndoorsLocationDataSource = IndoorsLocationDataSource(
+                                positioningFeatureTable,
+                                getFeatureTable("Pathways"),
+                                getFeatureTable("Levels"),
+                                positioningId
+                            )
+                            // start the location display (blue dot)
+                            startLocationDisplay()
+                        }
+                    }.onFailure {
+                        showError("The positioning table contain no data")
+                    }
+            }
+        } else {
+            showError("Positioning Table not found in FeatureTables")
+        }
```

Comment:
makes sense, but everything else like the creation of queryparameters and the indoorsLocationDataSource object, still needs to be under the if (featureTable.tableName == "IPS_Positioning") condition as there could be other feature tables as well.

Reply:
Updated it.

Reply:
I don't think an extra condition is needed because if `featureTable.tableName != "IPS_Positioning"` is false then the table name must be `IPS_Positioning`, so there's no need to add an extra check for `featureTable.tableName == "IPS_Positioning"` because it will always be true.

Either way I think it's better to go with the suggestion in https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/80/files/836e3d07f4684c8f6f938b8b2635784d68ef7373#r1146930505 instead.